odk1-src/form-functions.rst

```diff
@@ -0,0 +1,502 @@
+
+
+.. function:: regex(string, expression)
+
+  Returns ``True`` if ``string`` is an exact match for ``expression``.
+
+  .. seealso:: :doc:`form-regex`
+
+
+.. function:: contains(haystack, needle)
+
+  Returns ``True`` if the string ``haystack`` contains the substring ``needle``.
+
+.. function:: starts-with(haystack, needle)
+
+  Returns ``True`` if the string ``haystack`` begins with the substring ``needle``.
+
+.. function:: ends-with(haystack, needle)
+
+  Returns ``True`` if the string ``haystack`` ends with the substring ``needle``.
+
+
+.. function:: substr(string, start[, end]) 	
+
+  Returns the substring of ``string`` beginning at the index ``start`` and extending to (but not including) index ``end`` (or to the termination of ``string``, if ``end`` is not provided). Members of ``string`` are zero-indexed.
+  
+.. function:: string-length(string)
+
+  Returns the number of characters in ``string``.
+  
+  
+Combining strings
+~~~~~~~~~~~~~~~~~~  
+
+.. function:: concat(arg [, arg [, arg [, arg [...]]]])
+
+  Concatenates one or more arguments into a single string. If any ``arg`` is a :term:`nodeset`, the values within the set are concatenated into a string.
+
+  
+.. function:: join(separator, nodeset)
+
+  Joins the members of `nodeset`, using the string `separator`.
+
+  
+Converting to and from strings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. function:: boolean-from-string(string)
+
+  Returns ``True`` if ``string`` is "true" or "1".
+  Otherwise, ``False``.
+
+
+.. function:: number(string)
+
+  Converts a ``string`` of digits into a number value.
+
+.. function:: string(arg)
+
+   Converts ``arg`` to a string.
+
+
+  
+Math and numbers
+------------------
+
+
+.. function:: round(number, places)
+
+  Rounds a decimal ``number`` to some number of decimal ``places``.
+
+.. function:: int(arg) 	
+
+  Converts ``arg`` to an integer.
+
+
+.. function:: pow(value, power)
+
+  Raises a ``value`` to a ``power``.
+
+.. function:: log(number)
+
+  Returns the natural log of ``number``.
+
+.. function:: log10(number)
+
+  Returns the base-10 log of ``number``.
+
+.. function:: abs(number)
+
+  Returns the absolute value of ``number``.
+
+.. function:: sin(number)
+
+  Returns the sine of ``number``.
+
+.. function:: cos(number)
+
+  Returns the cosine of ``number``.
+  
+.. function:: tan(number)
+
+  Returns the tanget of ``number``.
+
+.. function:: asin(number)
+
+  Returns the arc sine of ``number``.
+  
+.. function:: acos(number)
+
+  Returns the arc cosine of ``number``.
+
+.. function:: atan(number)
+
+  Returns the arctan of ``number``.
+
+.. function:: atan2(y,x)
+
+  Returns the multi-valued inverse tangent of ``y``, ``x``.
+
+.. function:: sqrt(number) 
+
+  Returns the square root of ``number``.
+
+.. function:: exp(x) 
+
+  Returns ``e^x``.
+
+.. function:: exp10(number arg)
+
+  Returns ``10^x``.
+
+.. function:: pi()
+
+  Returns an approximation of the mathematical constant π.
+
+
+Control flow
```

Comment:
This feels out of place here. Maybe it should go first or second (after form data)